### PR TITLE
Main nav: Update product menu

### PR DESCRIPTION
**Pull request by Yndira-E**
## Description

Updated the product menu to include device agent, project nodes and group items accordingly

## Related Issue(s)

[#2275](https://github.com/FlowFuse/website/issues/2275)

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by Yndira-E** [comment]
I didn't remove the roadmap. I don't see why it would be beneficial, and it's the only link we have to the changelog. However, I would be open to changing it to link directly to the changelog if we decide the roadmap page adds no value.

**Comment by joepavitt** [comment]
Changing it to Changelog would be a great idea imo.

The "Roadmap" is not maintained and makes us look a little poorly planned. The changeling shows activity, and is updated with shiny new things regularly

**Comment by Yndira-E** [comment]
> The "Roadmap" is not maintained

Just to clarify, it pulls the issues from GH, do you mean it's not maintained on GH? Is it pulling from the wrong place? Or did you think it had to be updated manually?

File: src/_includes/layouts/base.njk [hunk 1]
```diff
@@ -188,14 +188,22 @@ eleventyComputed:
...
+                        <ul class="narrow sm:grid sm:grid-flow-col sm:grid-rows-6 sm:pr-1 align-left sm:auto-rows-auto items-center">
+                            {% navoption "FlowFuse", null, 1, null, true, "pl-3 title border-l-2" %}{% endnavoption %}
+                            <ul class="sub-menu grid grid-rows-subgrid row-span-3 ml-7 auto-rows-auto border-l-2">
+                                {% navoption "Features", "/product/features/", 2, "star" %}{% endnavoption %}
+                                {% navoption "FlowFuse Cloud", "/product/cloud/", 2, "cloud" %}{% endnavoption %}
+                                {% navoption "Roadmap", "/product/roadmap/", 2, "map" %}{% endnavoption %}
+                            </ul>
+                            {% navoption "Customer Stories", "/customer-stories/", 1, "presentation" %}{% endnavoption %}
+                            {% navoption "Blueprint Library", "/blueprints/", 1, "building-library" %}{% endnavoption %}
+                            {% navoption "Extras", null, 1, null, true, "pl-3 title border-l-2" %}{% endnavoption %}
```
Comment (ZJvandeWeg) [inline src/_includes/layouts/base.njk:200]: Extra's? That's a bit weird IMO, and devalues the items below

**Comment by joepavitt** [comment]
> Do you mean it's not maintained on GH? Is it pulling from the wrong place? Or did you think it had to be updated manually?

It's running from an old project board that is no longer actively maintained: https://github.com/orgs/FlowFuse/projects/5 and in fact, uses a soon-to-be-deprecated GH project type too.

> Extra's? That's a bit weird IMO, and devalues the items below

Good point, not sure of an alternative phrasing? Or we don't put them into a sub-section, and surface it all at the same level?

**Comment by ZJvandeWeg** [comment]
Not calling them extra's, but components? Does that work? These are all elements of what's needed to provide our end to end data service, but the word 'services' doesn't fit for me.

**Comment by joepavitt** [comment]
> These are all elements of what's needed to provide our end to end data service, but the word 'services' doesn't fit for me.

The other term I'd used in a blog was "offerings"? But it feels odd as a heading in a navigation menu

**Comment by Yndira-E** [comment]
How about:

**Product:**
- **Platform**
  - Features
  - FlowFuse Cloud
  - Changelog
- Customer Stories
- Blueprints
- **Components**
  - Dashboard
  - Device Agent
  - Project Nodes
  
I changed the title "FlowFuse" to "Platform" because it seemed to me that what's outside of it is not FlowFuse... Maybe Blueprints should also be under Platform? And I think Components could work.

**Comment by joepavitt** [comment]
Yep, agree with that Yndira, also agree: blueprints

**Comment by Yndira-E** [comment]
I implemented the changes yesterday, including moving Blueprints under Platform. Let me know if anything else needs adjustment. Thanks!

**Comment by joepavitt** [comment]
Feels a little odd having "Customer Stories" on the right, I know it balances the menu, but it gets a little lost, how about having it underneath the "Product" like this instead?:

Screenshot 2024-07-17 at 11 56 48

**Comment by Yndira-E** [comment]
Yeah, I had it like that at first, but it looks out of balance and still feels off, like the orphan one, hehe. So, I thought to place it to the right; it would still be the orphan but at least balanced... I even thought about removing it, we also have a link to customer stories under resources.

**Pull request merged**